src/ReverseProxy/Service/Proxy/Infrastructure/ProxyHttpClientFactory.cs

```diff
@@ -12,66 +14,85 @@ namespace Microsoft.ReverseProxy.Service.Proxy.Infrastructure
     /// </summary>
     internal class ProxyHttpClientFactory : IProxyHttpClientFactory
     {
-        /// <summary>
-        /// Handler for http requests.
-        /// </summary>
-        private readonly HttpMessageHandler _handler;
-
-        private bool _disposed;
+        private readonly ILogger<ProxyHttpClientFactory> _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ProxyHttpClientFactory"/> class.
         /// </summary>
-        public ProxyHttpClientFactory()
+        public ProxyHttpClientFactory(ILogger<ProxyHttpClientFactory> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <inheritdoc/>
+        public HttpMessageInvoker CreateClient(ProxyHttpClientContext context)
         {
-            _handler = new SocketsHttpHandler
+            if (CanReuseOldClient(context))
+            {
+                Log.ProxyClientReused(_logger, context.ClusterID);
+                return context.OldClient;
+            }
+
+            var newClientOptions = context.NewOptions;
+            var handler = new SocketsHttpHandler
             {
                 UseProxy = false,
                 AllowAutoRedirect = false,
                 AutomaticDecompression = DecompressionMethods.None,
-                UseCookies = false,
-                MaxConnectionsPerServer = int.MaxValue, // Proxy manages max connections
+                UseCookies = false
 
                 // NOTE: MaxResponseHeadersLength = 64, which means up to 64 KB of headers are allowed by default as of .NET Core 3.1.
             };
-        }
 
-        /// <inheritdoc/>
-        public HttpMessageInvoker CreateClient()
-        {
-            if (_disposed)
+            if (newClientOptions.SslProtocols != null)
+            {
+                handler.SslOptions.EnabledSslProtocols = newClientOptions.SslProtocols.Value;
+            }
+            if (newClientOptions.ClientCertificate != null)
+            {
+                handler.SslOptions.ClientCertificates = new X509CertificateCollection
+                {
+                    newClientOptions.ClientCertificate
+                };
+            }
+            if (newClientOptions.MaxConnectionsPerServer != null)
             {
-                throw new ObjectDisposedException(typeof(ProxyHttpClientFactory).FullName);
+                handler.MaxConnectionsPerServer = newClientOptions.MaxConnectionsPerServer.Value;
+            }
+            if (!newClientOptions.ValidateRemoteCertificate)
+            {
+                handler.SslOptions.RemoteCertificateValidationCallback = delegate { return true; };
             }
 
-            return new HttpMessageInvoker(_handler, disposeHandler: false);
+            Log.ProxyClientCreated(_logger, context.ClusterID);
+            return new HttpMessageInvoker(handler, disposeHandler: false);
```

Comment:
The handler shouldn't outlive the invoker anymore.
```suggestion
            return new HttpMessageInvoker(handler, disposeHandler: true);
```